dev-docs/RFCs/v7.x/post-process-effect-rfc.md

```diff
@@ -0,0 +1,66 @@
+# RFC: Post-processing Effect Rendering
+
+* **Authors**: Jian Huang
+* **Date**: May. 2019
+* **Status**: For Review
+
+## Abstract
+Post-processing effect rendering is part of deck.gl effect rendering pipeline, focusing on screen space pixel manipulation by using ping-pong technique.
+
+## Design
+
+A post-processing effect is created from **PostProcessEffect** class and **Shader module**.
+```js
+import {Convolution} from @luma.gl/Effects;
+import {PostProcessEffect} from @deck.gl/core;
+
+const postProcessEffect = new PostProcessEffect(Convolution, {  
+  kernel: Convolution.KERNEL.EMBOSS,  
+  kernelWeight: Convolution.KERNEL.EMBOSS.reduce((sum, x) => sum + x, 0)  
+});
+```
+
+### API
+
+PostProcessEffect class is the public interface to create post-processing effects, it only has two params.
+* module (Object): GLSL shader code and default uniforms for rendering
+* props (Object): parameters to tune effects
+
+ScreenPass class is the internal interface to handle the rendering of one pass, it is created by PostProcessEffect dynamically from shader module.
+* inputBuffer(Object): frame buffer object as input data
+* outputBuffer(Object): frame buffer object as out
+
+### Shader Module
+@luma.gl/Effects lib will host all the post-processing effect shader modules, currently most of them are sitting in @luma.gl/glfx. A typical shader module defines **name**, **uniforms** and **Shader Func**.
```

Comment:
Why do we need to make a new module? Does the old module go away? What are the differences between the two?

Reply:
For now most of the effect shader modules are sitting inside @luma.gl/glfx, better to have a new module to host them, and in glfx we import new module and use. We already have a @luma.gl/Effects module which holds passes and shader modules, going to refactor it to hold new shader modules from glfx.